jax/_src/lax/control_flow.py

```diff
@@ -205,8 +206,8 @@ def fori_loop(lower, upper, body_fun, init_val):
   if (isinstance(core.get_aval(lower), ConcreteArray) and
       isinstance(core.get_aval(upper), ConcreteArray)):
     try:
-      lower_ = int(lower)
-      upper_ = int(upper)
+      lower_ = _int_replaced_by_index(lower)
+      upper_ = _int_replaced_by_index(upper)
```

Comment:
The intent here is to check whether `lower` and `upper` are tracers. We should not raise deprecation warnings here, and we probably should not change this to `operator.index` in the long run.

Reply:
Good catch! This is my mistake!

Reply:
> The side effect here is that non-traced floating point array inputs would be switched from using the scan implementation to using the while_loop.implementation, which is not what we want.

What about using concrete check here?